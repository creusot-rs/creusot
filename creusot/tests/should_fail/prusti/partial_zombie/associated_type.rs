extern crate creusot_contracts;
use creusot_contracts::prusti_prelude::*;

trait T {
    type A;
}

impl T for u32 {
    type A = Box<u32>;
}

impl T for Box<u32> {
    type A = u32;
}

impl<A: T, B: T> T for (A, B) {
    type A = (A::A, B::A);
}
type A<X> = <X as T>::A;
#[ensures(x == x)]
fn test<X: T>(x: (A<X>, A<u32>, A<Box<u32>>, A<(X, u32)>)) {}
