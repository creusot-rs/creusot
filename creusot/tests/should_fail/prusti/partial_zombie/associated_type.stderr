error: binary operation `==` cannot be applied to type `(creusot_contracts::prusti::Zombie<<X as T>::A>, creusot_contracts::prusti::Zombie<std::boxed::Box<u32>>, u32, (creusot_contracts::prusti::Zombie<<X as T>::A>, creusot_contracts::prusti::Zombie<std::boxed::Box<u32>>))` since it doesn't implement `SnapEq`
  --> associated_type.rs:20:11
   |
20 | #[ensures(x == x)]
   |           ^

error: aborting due to 1 previous error

